scio-extra/src/main/scala/com/spotify/scio/extra/Sparkey.scala

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2017 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.extra
+
+import java.io._
+import java.net.URI
+import java.nio.ByteBuffer
+import java.nio.file.{Paths, Files => JFiles}
+
+import com.google.common.io.Files
+import com.spotify.scio.ScioContext
+import com.spotify.scio.values.{SCollection, SideInput}
+import com.spotify.sparkey.{CompressionType, SparkeyReader, SparkeyWriter, Sparkey => JSparkey}
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions
+import org.apache.beam.sdk.options.PipelineOptionsFactory
+import org.apache.beam.sdk.transforms.{DoFn, View}
+import org.apache.beam.sdk.util.GcsUtil.GcsUtilFactory
+import org.apache.beam.sdk.util.gcsfs.GcsPath
+import org.apache.beam.sdk.values.PCollectionView
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+object Sparkey {
+
+  implicit class SparkeyScioContext(val self: ScioContext) {
+    def sparkeySideInput(url: SparkeyUri): SideInput[SparkeyReader] = {
+      val view = self.parallelize(Seq(url)).applyInternal(View.asSingleton())
+      new SparkeySideInput(view)
+    }
+  }
+
+  implicit class SCollectionWithSparkeyWriter[K, V](val self: SCollection[(K, V)])
+                                                   (implicit ev1: K <:< String, ev2: V <:< String) {
+    def asSparkey(uri: SparkeyUri): SCollection[SparkeyUri] = self.transform { in =>
+      in.groupBy(_ => ())
+        .map { case (_, iter) =>
+          val writer = uri.getWriter()
```

Comment:
Actually this is the only place `getWriter` is used. Get rid of the method, `WrappedSparkeyWriter` and 2 subclasses, and put everything here in a utility object? That's a lot cleaner.